components/search/FacetDropdown.vue

```diff
@@ -0,0 +1,253 @@
+
+  export default {
+    directives: {
+      clickOutside: vClickOutside.directive
+    },
+
+    props: {
+      facet: {
+        type: Object,
+        required: true
+      },
+
+      selectedFacet: {
+        type: [Array, String],
+        required: false,
+        default: () => []
+      },
+
+      facetType: {
+        type: String,
+        required: true
+      }
+    },
+
+    data() {
+      return {
+        RADIO: 'radio',
+        THEME: 'THEME',
+        buttonNames: {
+          THEME: this.$t('facets.THEME.name'),
+          TYPE: this.$t('facets.TYPE.name'),
+          REUSABILITY: this.$t('facets.REUSABILITY.name'),
+          COUNTRY: this.$t('facets.COUNTRY.name')
+        },
+        preSelected: [],
+        radioSelected: null
+      };
+    },
+
+    computed: {
+      sortOptions() {
+        let newArray = [].concat(this.facet.fields);
+        const selected = [];
+        let leftOver;
+
+        /* THIS NEEDS TO BE CLEANED UP */
+        newArray.map(field => {
+          if (this.selectedFacet.includes(field.label)) {
+            selected.push(field);
+          }
+        });
+
+        leftOver = newArray.filter(field => !this.selectedFacet.includes(field.label));
+
+        return selected.sort((a, b) => a.count + b.count).concat(leftOver);
+        /* END - THIS NEEDS TO BE CLEANED UP */
+      },
+
+      facetName() {
+        return this.buttonNames[this.facet.name];
+      },
+
+      isRadio() {
+        return this.facetType === this.RADIO;
+      },
+
+      activateCheckboxResetButton() {
+        return this.preSelected.length > 0;
+      },
+
+      disableRadioResetButton() {
+        return !this.radioSelected;
+      },
+
+      disableApplyButton() {
+        return this.preSelected.length === this.selectedFacet.length;
+      },
+
+      hasSelection() {
+        return (this.radioSelected || this.selectedFacet.length > 0) ? 'secondary' : 'light';
+      }
+    },
+
+    mounted() {
+      if (this.facetType === this.RADIO) {
+        if (Array.isArray(this.selectedFacet)) {
+          this.radioSelected = '';
+        } else {
+          this.radioSelected = this.selectedFacet;
+        }
+      } else if (this.selectedFacet.length > 0) {
+        this.preSelected = this.selectedFacet;
+      }
+    },
+
+    methods: {
+      resetCheckboxSelection() {
+        this.preSelected = [];
+      },
+
+      resetRadioSelection() {
+        this.radioSelected = '';
+      },
+
+      applySelection() {
+        if (this.isRadio) {
+          this.$emit('updated', this.THEME, this.radioSelected);
+          this.$refs.dropdown.hide(true);
+        } else {
+          this.selected = this.preSelected;
+          this.$emit('updated', this.facet.name, this.selected);
+          this.$refs.dropdown.hide(true);
+        }
+      }
+    }
+  };
+</script>
+
+<style lang="scss" scoped>
+  @import "./assets/scss/variables.scss";
+
+  .dropdown {
+    width: 100%;
```

Comment:
Alternative: you can use a Bootstrap utility class, "w-100".

Reply:
I can do, but since I have already added styling to the dropdown, I think it's better to not clog up the HTML too much with helper classes. If there was no `.dropdown` class then I would agree. Thoughts?